src/main/java/edu/byu/cs/autograder/QualityAnalyzer.java

```diff
@@ -66,67 +74,100 @@ private void parseOutput(String output) {
             StringBuilder categoryResultsBuilder = new StringBuilder();
-            for(String reporter : category.reporters()) {
+            for (String reporter : category.reporters()) {
                 StringBuilder reporterResultsBuilder = new StringBuilder();
-                if (errors.containsKey(reporter)) {
-                    errors.get(reporter).forEach(s -> reporterResultsBuilder.append("\t\t").append(s).append("\n"));
+                if (output.errors().containsKey(reporter)) {
+                    output.errors().get(reporter)
+                            .forEach(s -> reporterResultsBuilder.append("\t\t").append(s).append("\n"));
                 }
                 if (!reporterResultsBuilder.isEmpty()) {
-                    categoryResultsBuilder.append("\t").append(reporter).append(":\n")
-                            .append(reporterResultsBuilder);
+                    categoryResultsBuilder.append("\t").append(reporter).append(":\n").append(reporterResultsBuilder);
                 }
             }
             if (!categoryResultsBuilder.isEmpty()) {
                 resultsBuilder.append(category.name()).append(":\n").append(categoryResultsBuilder);
             }
         }
 
-        if(!warnings.isEmpty()) {
+        if (!output.warnings().isEmpty()) {
             resultsBuilder.append("Warnings:\n");
-            warnings.forEach(s -> resultsBuilder.append("\t").append(s).append("\n"));
+            output.warnings().forEach(s -> resultsBuilder.append("\t").append(s).append("\n"));
         }
 
         if (resultsBuilder.isEmpty()) resultsBuilder.append("Good job!");
         else resultsBuilder.deleteCharAt(resultsBuilder.length() - 1);
         return resultsBuilder.toString();
     }
 
-    private String getNotes() {
+    /**
+     * Creates a string with the output of each category
+     *
+     * @param output errors and warnings from quality analysis
+     * @return a String of displaying each category and if it passed or not
+     */
+    private String getNotes(QualityOutput output) {
         StringBuilder builder = new StringBuilder();
         for (QualityRubricCategory category : qualityRubricItems.categories()) {
-            boolean categoryPasses = categoryPasses(category);
+            boolean categoryPasses = categoryPasses(category, output);
             builder.append((categoryPasses) ? "✓" : "✗").append(" ").append(category.name()).append("\n");
         }
         return builder.toString();
     }
 
-
-    private boolean categoryPasses(QualityRubricCategory category) {
-        for(String reporter : category.reporters()) {
-            if(errors.containsKey(reporter)) return false;
+    /**
+     * Determines if a particular category had any quality errors
+     *
+     * @param category category to check
+     * @param output   errors and warnings from quality analysis
+     * @return false if an error was present for the category, false otherwise
```

Comment:
Did you mean to say `... true otherwise` at the end of this line?

Reply:
I did. Thank you.

Reply:
Lol we both committed to resolve this issue at the same time 😆 You pushed before I did though @19mdavenport...

<img width="978" alt="image" src="https://github.com/softwareconstruction240/autograder/assets/19808440/543a7223-72ec-40d5-8fdf-cf281d8f890f">